src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/GenericMethodsTemplateMap.cs

```diff
@@ -51,29 +51,23 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             nativeSection.Place(hashtable);
 
 
-            foreach (MethodDesc method in factory.MetadataManager.GetCompiledMethods())
```

Comment:
Going over compiled method bodies was flawed because we have cases where we need native layout for dictionaries of methods that weren't compiled. It has to do with the fact that we compute generic dictionary layouts without considering the effects of inlining - we can end up with generic dictionaries that have dead references to other generic dictionaries of methods that got inlined.

`Unsafe.As` is very famous for being in this category. Can't wait for it to be a RyuJIT intrinsic so that I can intrinsify it in the scanner too.